d3a_interface/sim_results/kpi.py

```diff
@@ -145,11 +145,74 @@ def update_area_kpi(self, area_dict, core_stats):
         self._accumulate_energy_trace(core_stats)
 
 
+class SavingsKPI:
+    def __init__(self):
+        self.producer_ess_set = set()
+        self.consumer_ess_set = set()
+        self.house_list = list()
+        self.fit_revenue = 0.
+        self.utility_bill = 0.
+        self.base_case_revenue = 0.
+        self.d3a_revenue = 0.
+        self.saving_absolute = 0.
+        self.saving_percentage = 0.
+
+    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
+        """
```

Comment:
Nit: the first line of a docstring should be a short summary that describes the method. Here it is missing, so it might be good to add it. Lines below could give some more information; for example: what's the "saving kpi"?

Reply:
Besides this, it's great that you're documenting the arguments for this function 👍